ludwig/hyperopt/ray.py

```diff
@@ -0,0 +1,133 @@
+        for arg in space_required_args:
+            if arg in values:
+                space_input_args[arg] = values[arg]
+            else:
+                raise ValueError(
+                    "Parameter '{}' not defined for {}".format(arg, param))
+        config[param] = space_function(**space_input_args)
+    return config
+
+
+def update_tune_hyperopt_params_with_defaults(hyperopt_params):
+    set_default_value(hyperopt_params, "split", VALIDATION)
+    set_default_value(hyperopt_params, "output_feature", COMBINED)
+    set_default_value(hyperopt_params, "metric", LOSS)
+    set_default_value(hyperopt_params, "goal", MINIMIZE)
+
+
+class RayTuneExecutor:
+    def __init__(self, search_space, output_feature, metric, split, goal, **kwargs):
+        ray.init(ignore_reinit_error=True)
+        self.search_space = search_space
+        self.output_feature = output_feature
+        self.metric = metric
+        self.split = split
+        self.goal = goal
+        self.trial_id = 0
+        self.experiment_kwargs = {}
+
+    def _run_experiment(self, config):
+
+        self.trial_id += 1
+        hyperopt_dict = copy.deepcopy(self.experiment_kwargs)
+        modified_config = substitute_parameters(
+            copy.deepcopy(hyperopt_dict["config"]), config)
+        hyperopt_dict["config"] = modified_config
+        hyperopt_dict["experiment_name"] = f'{hyperopt_dict["experiment_name"]}_{self.trial_id}'
+
+        train_stats, eval_stats = run_experiment(**hyperopt_dict)
+        metric_score = eval_stats[self.output_feature][self.metric]
+
+        tune.report(parameters=str(config), metric_score=metric_score,
+                    training_stats=str(train_stats), eval_stats=str(eval_stats))
+
+    def execute(self,
+                config,
+                dataset=None,
+                training_set=None,
+                validation_set=None,
+                test_set=None,
+                training_set_metadata=None,
+                data_format=None,
+                experiment_name="hyperopt",
+                model_name="run",
+                # model_load_path=None,
+                # model_resume_path=None,
+                skip_save_training_description=False,
+                skip_save_training_statistics=False,
+                skip_save_model=False,
+                skip_save_progress=False,
+                skip_save_log=False,
+                skip_save_processed_input=False,
+                skip_save_unprocessed_output=False,
+                skip_save_predictions=False,
+                skip_save_eval_stats=False,
+                output_directory="results",
+                gpus=None,
+                gpu_memory_limit=None,
+                allow_parallel_threads=True,
+                use_horovod=None,
+                random_seed=default_random_seed,
+                debug=False,
+                **kwargs):
+
+        self.experiment_kwargs = dict(
```

Comment:
This will be replaced by [tune.with_parameters](https://docs.ray.io/en/master/tune/api_docs/execution.html#tune-with-parameters) functionality in ray v1.1.0